assets/model_monitoring/components/src/model_performance_metrics/data_reader.py

```diff
@@ -0,0 +1,76 @@
+from shared_utilities.io_utils import read_mltable_in_spark
+
+
+class DataReaderFactory:
+
+    def __init__(self):
+        """
+
+        """
+        self.default_reader = BaseTaskReader
+
+    def get_reader(self, task):
+        """
+
+        Args:
+            task:
+
+        Returns:
+
+        """
+        # We would need different readers for tasks like text-generation(code)
+        return self.default_reader(task)
+
+
+class MetricsDTO:
+
+    def __init__(self, ground_truth, predictions):
+        self.ground_truth = ground_truth
+        self.predictions = predictions
+
+
+class BaseTaskReader:
+    """
+
+    """
+
+    def __init__(self, task):
+        """
+
+        Args:
+            task:
+        """
+        self.task = task
+
+    def _read_mltable(self, file_name, columns):
```

Comment:
nit: mind renaming this to something like `_read_mltable_to_pd_dataframe` to be explicit?